qiskit_dynamics/pulse/pulse_to_signals.py

```diff
@@ -65,14 +92,17 @@ def get_signals(self, schedule: Schedule) -> List[DiscreteSignal]:
             qiskit.QiskitError: if not enough frequencies supplied
         """
 
-        if self._carriers and len(self._carriers) < len(schedule.channels):
-            raise QiskitError("Not enough carrier frequencies supplied.")
+        if self._channels is None:
+            if self._carriers and len(self._carriers) < len(schedule.channels):
+                raise QiskitError("Not enough carrier frequencies supplied.")
 
         signals, phases, frequency_shifts = {}, {}, {}
 
         for idx, chan in enumerate(schedule.channels):
```

Comment:
If `self._channels is not None`, can we already start doing some "filtering" in these loops? E.g. only adding `chan.name` as a key to these `dict`s if `self._channels is not None` and `chan.name in self._channels` (similarly for the subsequent loop, if an instruction isn't associated to one of these channels, just skip to the next instruction.

I guess this isn't an option if there's some interdependency between channels/instructions? I know control channels have their frequency specified as linear combinations of frequencies of drive channels (unless this has changed since I last looked at it). If you set or shift frequency of a drive channel mid-schedule, does that impact the frequency of associated control channels?